controllers/nodehealthcheck_controller.go

```diff
@@ -532,3 +555,69 @@ func updateResultNextReconcile(result *ctrl.Result, updatedRequeueAfter time.Dur
 		result.RequeueAfter = updatedRequeueAfter
 	}
 }
+
+func (r *NodeHealthCheckReconciler) addWatch(remediationCR *unstructured.Unstructured) error {
+	key := remediationCR.GroupVersionKind().String()
+	if _, exists := r.watches[key]; exists {
```

Comment:
I think we need to make sure to access r.watches in a way that is concurrent safe

Reply:
fixed